[Game] Loot items for quests now only drop if the killer has the quest

ZeromusXYZ opened the pull request:
Fixes #1096 [BUG] Quest items obtained by cutting willows

Added code to check if the killer/user has the related quest active when looting. Also added helper function to propagate the killer to a player if needed.

This should also fix several other quest loot drops.

ZeromusXYZ committed:
[Game] Loot items for quests now only drop if the killer has the quest

Fixes #1096 [BUG] Quest items obtained by cutting willows

Added code to check if the killer/user has the related quest active when looting. Also added helper function to propagate the killer to a player if needed.

rogerbarreto commented on AAEmu.Game/Models/Game/Items/Loots/LootPack.cs:
```diff
@@ -155,9 +167,46 @@ public class LootPack
+    /// </summary>
+    /// <param name="killer">Unit doing the killing blow</param>
+    /// <returns></returns>
+    private Character GetPlayerUsingKiller(BaseUnit killer)
```
```suggestion
    private ICharacter GetPlayerUsingKiller(BaseUnit killer)
```

ZeromusXYZ committed:
Update AAEmu.Game/Models/Game/Items/Loots/LootPack.cs

Co-authored-by: Roger Barreto <19890735+RogerBarreto@users.noreply.github.com>

ZeromusXYZ committed:
[Game] Loot items for quests now only drop if the killer has the quest

Additional changes to make `GetPlayerUsingKiller` use `ICharacter` instead of `Character`
